File: frontend/src/types/broadcast.ts
export interface BroadcastRequest {
  group_ids: number[];
  message_type: 'auto_generate' | 'custom';
  table_group_ids?: number[];
  custom_message?: string;
  send_immediately: boolean;
  scheduled_hour?: number;
  scheduled_minute?: number;
}

export interface BroadcastResult {
  history_id?: number;
  group_id: number;
  group_name: string;
  status: string;
  scheduled_time?: string;
  error?: string;
}

export interface BroadcastResponse {
  message: string;
  results: BroadcastResult[];
}

export interface BroadcastHistory {
  id: number;
  group_id: number;
  group_name: string;
  message_preview: string;
  message_type: string;
  table_groups?: string[];
  scheduled_for: string;
  sent_at?: string;
  status: 'pending' | 'scheduled' | 'sent' | 'failed';
  error_message?: string;
  created_at: string;
}

export interface BroadcastHistoryList {
  history: BroadcastHistory[];
  total_count: number;
}

File: backend/schemas/broadcast.py
from pydantic import BaseModel, validator
from datetime import datetime
from typing import Optional, List

class BroadcastRequest(BaseModel):
    group_ids: List[int]
    message_type: str  # 'auto_generate' or 'custom'
    table_group_ids: Optional[List[int]] = None
    custom_message: Optional[str] = None
    send_immediately: bool = True
    scheduled_hour: Optional[int] = None
    scheduled_minute: Optional[int] = None
    
    @validator('message_type')
    def validate_message_type(cls, v):
        if v not in ['auto_generate', 'custom']:
            raise ValueError('message_type must be auto_generate or custom')
        return v
    
    @validator('table_group_ids')
    def validate_auto_generate(cls, v, values):
        if values.get('message_type') == 'auto_generate' and not v:
            raise ValueError('table_group_ids required for auto_generate')
        return v
    
    @validator('custom_message')
    def validate_custom(cls, v, values):
        if values.get('message_type') == 'custom' and not v:
            raise ValueError('custom_message required for custom type')
        return v

class BroadcastResult(BaseModel):
    history_id: Optional[int] = None
    group_id: int
    group_name: str
    status: str
    scheduled_time: Optional[datetime] = None
    error: Optional[str] = None

class BroadcastResponse(BaseModel):
    status: str
    message: str
    results: List[BroadcastResult]

class BroadcastHistoryResponse(BaseModel):
    id: int
    group_id: int
    group_name: str
    message_preview: str
    message_type: str
    table_groups: Optional[List[str]] = None
    scheduled_for: datetime
    sent_at: Optional[datetime] = None
    status: str
    error_message: Optional[str] = None
    created_at: datetime
    
    class Config:
        from_attributes = True
